mypy/checker.py

```diff
@@ -1483,6 +1486,19 @@ def check_assignment(self, lvalue: Lvalue, rvalue: Expression, infer_lvalue_type
         else:
             lvalue_type, index_lvalue, inferred = self.check_lvalue(lvalue)
 
+            if isinstance(lvalue, NameExpr):
```

Comment:
Rearranged the code a bit. I think the only logic change is that we may call `accept` on the `rvalue` more than once, which is probably fine because errors get deduplicated somewhere.

Relatedly, the way this method is structured makes it hard to verify that we actually call `accept` on the rvalue at all times. It might be worth it to refactor the code to just call `self.expr_checker.accept(rvalue)` once at the top.